src/compile/Model.ts

```diff
@@ -233,48 +233,88 @@ export class Model {
     return 30;
   }
 
-  /** @return an object with format and formatType properties. */
+  /**
+   * Returns an object with format and formatType properties.
+   */
   public formatMixins(channel: Channel, format: string) {
     const fieldDef = this.fieldDef(channel);
 
     let def: any = {};
 
-    if (fieldDef.type === TEMPORAL) {
-      // explicitly set the fromat type so that vega uses the datetime formatter
-      def.formatType = 'time';
+    switch (fieldDef.type) {
+      case QUANTITATIVE:
+        def.formatType = 'number';
+        break;
+      case TEMPORAL:
+        def.formatType = 'time';
+        break;
     }
 
     if (format !== undefined) {
       def.format = format;
-      return def;
+    } else {
+      switch (fieldDef.type) {
+        case QUANTITATIVE:
+          def.format = this.numberFormat(channel);
+          break;
+        case TEMPORAL:
+          const f = this.timeFormat(channel);
+          if (f) {
+            def.format = f;
+          } else {
+            def.format = this.config().timeFormat;
+          }
+          break;
+      }
     }
 
-    switch (fieldDef.type) {
-      case QUANTITATIVE:
-        def.format = this.numberFormat(channel);
-        break;
-      case TEMPORAL:
-        const f = this.timeFormat(channel);
-        if (f) {
-          def.format = f;
-        } else {
-          def.format = this.config().timeFormat;
+    if (def.formatType && channel === TEXT) {
+      // text does not support format and formatType
+      // https://github.com/vega/vega/issues/505
+
+      const filter = def.formatType + (def.format ? ':\'' + def.format + '\'' : '');
+      return {
+        text: {
+          template: '{{' + this.field(channel, {datum: true}) + ' | ' + filter + '}}'
         }
-        break;
+      };
+    }
+
+    if (def.formatType === 'number') {
```

Comment:
Better have 

`if ( contains([QUANTITATIVE, TEMPORAL], fieldDef.type) {`  

surrounds all logic in this function.  (`format` should be dropped for nominal and ordinal anyway!)

and do
```
if (fieldDef.type === TEMPORAL) {
   def.formatType = 'time'; 
}
```